src/App.js

```diff
@@ -1,25 +1,159 @@
-import logo from './logo.svg';
-import './App.css';
+import React, { useState } from "react";
+import "./SlotMachine.css";
+
+const symbols = ["🍍", "🐠", "🥥", "🌴", "🐒", "🏴‍☠️"];
+const credits = [1, 5, 10, 20, 50, 100];
+
+function getRandomSymbol() {
+  return symbols[Math.floor(Math.random() * symbols.length)];
+}
+
+const SlotMachine = () => {
+  const [reels, setReels] = useState(
+    Array(3)
+      .fill(null)
+      .map(() => Array(3).fill(symbols[2]))
+  );
+  const [isSpinning, setIsSpinning] = useState(false);
+  const [message, setMessage] = useState("Deposit to play!");
+  const [balance, setBalance] = useState(100);
+  const [bet, setBet] = useState(credits[0]);
+  const [spins, setSpins] = useState(0);
+  const [wins, setWins] = useState(0);
+  const [losses, setLosses] = useState(0);
+
+  // Spins the reels
+  const spinReels = () => {
+    if (bet > balance) {
+      setMessage("Insufficient balance to spin!");
+      return;
+    }
+
+    setIsSpinning(true);
+    setMessage("");
+    setBalance(balance - bet);
+    setSpins(spins + 1);
+
+    let spinCount = 0;
+    const spinInterval = setInterval(() => {
+      setReels(
+        Array(3)
+          .fill(null)
+          .map(() =>
+            Array(3)
+              .fill(null)
+              .map(() => getRandomSymbol())
+          )
+      );
+      spinCount++;
+      if (spinCount > 10) {
+        clearInterval(spinInterval);
+        finalizeSpin();
+      }
+    }, 100);
+  };
+
+  // Checks if the player won
+  const checkWin = (finalReels) => {
+    const winningLines = [
```

Comment:
This method is nicely encapsulated, so you can move it outside of the component and it can either live at the top of the file or in a utils file. That way, you can isolate it and test it. Also, whenever the component rerenders, it has to save this into memory again. So if you move it outside the component, even if the component rerenders, it'll already be saved into memory at the initial load. This saves computation time.